src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -75976,6 +75976,66 @@ static void F(E e)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "e").WithLocation(11, 29));
         }
 
+        [Fact]
+        public void OtherComparisonsAsPureNullTests_ExtendedProperties_NotNull()
+        {
+            var source =
+@"#nullable enable
+class E
+{
+    public E Property1 { get; set; } = null!;
+    public object Property2 { get; set; } = null!;
+}
+class Program
+{
+    static void F(E e)
+    {
+        switch (e)
+        {
+            case var x when e.Property1.Property2.ToString() == null: // 1
+                break;
+            case { Property1.Property2: not null }:
```

Comment:
>not nul [](http://example.com/codeflow?start=40&length=7)

Nit: consider just making this a theory with all the different pure tests (such as both `null` and `not null`.